tests/std/include/test_regex_support.hpp

```diff
@@ -116,6 +116,59 @@ class regex_fixture {
         }
     }
 
+    void should_capture_groups(const std::string& subject, const std::string& pattern,
+        const std::vector<std::string>& groups,
+        const std::regex_constants::syntax_option_type flags = std::regex_constants::ECMAScript) {
+        try {
+            const std::regex r(pattern, flags);
+            std::smatch m;
+
+            if (!std::regex_match(subject, m, r)) {
+                printf(R"(Expected regex("%s") to match "%s".)"
+                       "\n",
+                    pattern.c_str(), subject.c_str());
+                fail_regex();
+                return;
+            }
+
+            if (m[0] != subject) {
+                printf(R"(should_capture("%s", "%s"): not a full match)"
+                       "\nexpected: \"%s\""
+                       "\nfound:    \"%s\"\n",
+                    subject.c_str(), pattern.c_str(), subject.c_str(), m[0].str().c_str());
+                fail_regex();
+            }
+
+            if (m.size() != groups.size() + 1) {
+                printf(R"(should_capture("%s", "%s"): captures: %zd != %zd)"

```

Comment:
`size_t` is unsigned, so this should say `%zu` (as `should_capture()` does). Occurs below.